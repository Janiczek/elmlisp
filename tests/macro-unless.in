(define-syntax-rule

  ; from...
  (unless condition
    true-expr
    false-expr)

  ; ...to
  (if condition
    false-expr
    true-expr))

(unless (== 3 5)
  "three is not equal to five"
  "this shouldn't happen")

; this gets expanded into
;
; (if (== 3 5)
;   "this shouldn't happen"
;   "three is not equal to five")
;
; which gets compiled to Elm code.



