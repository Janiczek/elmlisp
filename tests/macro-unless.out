if 3 == 5 then
    "this shouldn't happen"
else
    "three is not equal to five"
